### PR TITLE
Add command to invalidate spawn locations in a dimension

### LINKED ISSUE #7: Add a mechanism for invalidating a player's dimensionCache
Players updating from DimensionalEdibles may have a bad cached location already saved to their Player entity NBT for their spawn point. While we fixed the spawning bug itself in #3, we need some manner to invalidate the cache so their destination in the dimension is recomputed.

Perhaps a command would be appropriate, like `/dimensionaledibles reset [dimID|all]`, which would clear the tag off the calling player for either the specified dimension or all dimensions.

### PR DESCRIPTION
**Pull request by ALongStringOfNumbers**
To tag on to #3 and closes #7, this PR adds a command to invalidate the stored dimPos of the spawn point in a dimension for the player. This is useful for when the player had an invalid spawn point before the changes that went in in #3.

The syntax is `/dimensionalEdibles invalidate <dimID>`

### CODE REVIEW

**Review by Exaxxion — CHANGES_REQUESTED** [review]
This looks like a great start. During review I realized a few things that we should take into consideration.

The first one is that server operators might want to be able to configure the permission level required to issue the command. Normally the place you teleport to with a cake is locked-in on first use, but being able to invalidate the cache means you can move the cake to end up somewhere else. This is a change from the prior behavior that could potentially be exploited, so this command could reasonably be considered a cheat for administration purposes.

On that same vein, if players don't have permission to use the command, it should be possible for an operator/admin to issue it on behalf of another player. We should ask for an additional argument which is the target player's name.

Lastly, the language file entry for the command failing seems to be almost the same as the usage string, except one has brackets around `dimID`. We should give the caller a more informative error message depending on what problem was encountered.

**Comment by ALongStringOfNumbers** [comment]
I have now added a required player argument to the command, which can be done with @p if you want to target yourself, and made the operator level to execute the command configurable. If the player is incorrect, the generic minecraft incorrect player message will be shown.

File: src/main/java/jackyy/dimensionaledibles/registry/ModConfig.java [hunk 1]
```diff
@@ -30,6 +30,10 @@
         public boolean overworldApple = true;
         @Config.Comment("Set to true to enable Custom Cake.")
```
Comment (aimiebell) [inline src/main/java/jackyy/dimensionaledibles/registry/ModConfig.java:31]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

Comment (Exaxxion) [reply src/main/java/jackyy/dimensionaledibles/registry/ModConfig.java:31]: I think you misunderstand what these are: nested Annotation classes defined by the Forge API. We have no control over that code and merely call them as required for correct integration with Forge config file persistence.

Comment (serenibyss) [reply src/main/java/jackyy/dimensionaledibles/registry/ModConfig.java:31]: I’m fairly certain this is a bot, based on their Github history. Gregicality has had similar reviews like this from the same account, and they were blocked.

Comment (htmlcsjs) [reply src/main/java/jackyy/dimensionaledibles/registry/ModConfig.java:31]: ~400 reviews in a day, must be a bot

**Review by Exaxxion — APPROVED** [review]

**Pull request merged**